docs-src/docs.md

```diff
@@ -0,0 +1,422 @@
+
+![image alt text](images/image_4.png)
+
+### Setup git ssh access
+
+This project uses ssh public/private key pairs in order to authenticate the jenkins user to git server. [Why?](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/faq.md#why-ssh-for-git-authentication)
+
+At the end of the image creation process, Dockerfile generates and prints out a new public key that you can setup to give jenkins push/pull access to your repo.
+
+If you pushed the project files to your GitHub account, then go to your repo and within Settings click Deploy Keys on the left-hand menu. Next Add deploy key Give it a meaningful name under Title, copy/paste the contents of the public key as a new key and enable Allow write access option. And finally hit Add key
+
+![image alt text](images/image_5.png)
+
+See [GitHub Deploy Keys Documentation](https://developer.github.com/guides/managing-deploy-keys/#deploy-keys)
+
+### Run image
+Positive
+: Note: remove any previous currency-v1 deployments from the Apigee org in order to demonstrate proxies created successfully with correct revision numbers.
+
+All parameters required to run Jenkins and Maven are supplied using environment variables. These variables are set directly to the docker container with the docker run command.
+
+There is a script provided to make this process easier which can be found [here](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/setup/run-image.sh). Modify this script to provide values specific to your setup and execute the script to run the image.
+
+Please note that setup/run-image.sh script is in source control. Copy this file to setup/.run-image.sh before making any changes to prevent sensitive data committed to this repo. Root .gitignore file contains a directive to ignore any changes to setup/.run-image.sh.
+
+Go to the cloned folder
+
+```
+cp ./setup/run-image.sh ./setup/.run-image.sh
+```
+
+Modify the file and plug in your environment data:
+
+```
+vim ./setup/.run-image.sh
+```
+
+
+You get that data from your Apigee Account. You can look it up by going to your helloworld proxy.
+
+![image alt text](images/image_6.png)
+
+Positive
+: Note: Also change the port to 8080 if you are working on a Chromebook as the port 9001 cannot be used by docker containers.
+
+You can also have to change the name to match the one of your docker images. Change apigee/ci to apigee/ci-demo-walkthrough and also replace my-ci with a unique name that does not match another container running on your machine.
+
+![image alt text](images/image_7.png)
+
+A Description of the fields is within the file. After done with the modification run the script.
+
+```
+. ./setup/.run-image.sh
+```
+
+Script will output the docker container hash.
+
+### Access Jenkins
+
+Fire up a browser and hit http://localhost:8080 to access Jenkins UI.
```

Comment:
lets say something like "your docker url or localhost". Often people run this on VMs so would need an IP address. Old versions of docker used virtualbox under the hood and required a different IP address too